src/main/java/com/authservice/config/UserInfoUserDetailsService.java

```diff
@@ -0,0 +1,23 @@
+package com.authservice.config;
+import com.authservice.model.User;
+import com.authservice.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+import java.util.Optional;
+
+@Component
+public class UserInfoUserDetailsService implements UserDetailsService {
+
+    @Autowired
+    private UserRepository repository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        Optional<User> userInfo = repository.findByName(username);
```

Comment:
findByEmail

Reply:
updated code changes; changed findByName(username)  to findByEmail(email)